src/Arcus.WebApi.Logging.AzureFunctions/Extensions/IFunctionsWorkerApplicationBuilderExtensions.cs

```diff
@@ -123,7 +137,10 @@ public static IFunctionsWorkerApplicationBuilder UseExceptionHandling<TMiddlewar
         public static IFunctionsWorkerApplicationBuilder UseRequestTracking(
             this IFunctionsWorkerApplicationBuilder builder)
         {
-            Guard.NotNull(builder, nameof(builder), "Requires a function worker builder instance to add the HTTP request tracking middleware");
+            if (builder is null)
```

Comment:
With overloads, you could remove the checks if the value it checks isn't used by the method itself. In the previous code, there were some extensive 'paranoia' guards.

Reply:
I'll reduce the amount of checks and attempt to centralize them.